vms/evm/sync/customrawdb/accessors_state_sync.go

```diff
@@ -0,0 +1,212 @@
+	start := keyBytes[common.HashLength:]
+	return root, start
+}
+
+// NewSyncStorageTriesIterator returns a KeyLength iterator over all storage tries
+// added for syncing (beginning at seek). It is the caller's responsibility to unpack
+// the key and call Release on the returned iterator.
+func NewSyncStorageTriesIterator(db ethdb.Iteratee, seek []byte) ethdb.Iterator {
+	return rawdb.NewKeyLengthIterator(db.NewIterator(syncStorageTriesPrefix, seek), syncStorageTriesKeyLength)
+}
+
+// WriteSyncStorageTrie adds a storage trie for account (with the given root) to be synced.
+func WriteSyncStorageTrie(db ethdb.KeyValueWriter, root common.Hash, account common.Hash) error {
+	bytes := make([]byte, 0, syncStorageTriesKeyLength)
+	bytes = append(bytes, syncStorageTriesPrefix...)
+	bytes = append(bytes, root[:]...)
+	bytes = append(bytes, account[:]...)
+	return db.Put(bytes, []byte{0x01})
+}
+
+// ClearSyncStorageTrie removes all storage trie accounts (with the given root) from db.
+// Intended for use when the trie with root has completed syncing.
+func ClearSyncStorageTrie(db ethdb.KeyValueStore, root common.Hash) error {
+	accountsPrefix := make([]byte, len(syncStorageTriesPrefix)+common.HashLength)
+	copy(accountsPrefix, syncStorageTriesPrefix)
+	copy(accountsPrefix[len(syncStorageTriesPrefix):], root[:])
+	return clearPrefix(db, accountsPrefix, syncStorageTriesKeyLength)
+}
+
+// ClearAllSyncStorageTries removes all storage tries added for syncing from db
+func ClearAllSyncStorageTries(db ethdb.KeyValueStore) error {
+	return clearPrefix(db, syncStorageTriesPrefix, syncStorageTriesKeyLength)
+}
+
+// UnpackSyncStorageTrieKey returns the root and account for a storage trie
+// key returned from NewSyncStorageTriesIterator.
+func UnpackSyncStorageTrieKey(keyBytes []byte) (common.Hash, common.Hash) {
+	keyBytes = keyBytes[len(syncStorageTriesPrefix):] // skip prefix
+	root := common.BytesToHash(keyBytes[:common.HashLength])
+	account := common.BytesToHash(keyBytes[common.HashLength:])
+	return root, account
+}
+
+// WriteSyncPerformed logs an entry in `db` indicating the VM state synced to `blockNumber`.
+func WriteSyncPerformed(db ethdb.KeyValueWriter, blockNumber uint64) error {
+	syncPerformedPrefixLen := len(syncPerformedPrefix)
+	bytes := make([]byte, syncPerformedPrefixLen+wrappers.LongLen)
+	copy(bytes[:syncPerformedPrefixLen], syncPerformedPrefix)
+	binary.BigEndian.PutUint64(bytes[syncPerformedPrefixLen:], blockNumber)
+	return db.Put(bytes, []byte{0x01})
+}
+
+// NewSyncPerformedIterator returns an iterator over all block numbers the VM
+// has state synced to.
+func NewSyncPerformedIterator(db ethdb.Iteratee) ethdb.Iterator {
+	return rawdb.NewKeyLengthIterator(db.NewIterator(syncPerformedPrefix, nil), syncPerformedKeyLength)
+}
+
+// UnpackSyncPerformedKey returns the block number from keys the iterator returned
+// from NewSyncPerformedIterator.
+func UnpackSyncPerformedKey(key []byte) uint64 {
```

Comment:
super-nit: I'm not sure when we use `Unpack*` naming or why we started using it... but I feel like now would be a reasonable time to stop using it and use something like `Parse` or `Unmarshal` which we use more commonly